### PR TITLE
Add the missing loaders tests

### PR DESCRIPTION
**Pull request by RaananW**
And cleaning up the old ones

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/loaders/test/integration/babylon.sceneLoader.test.ts [hunk 1]
```diff
@@ -197,10 +197,960 @@ describe("Babylon Scene Loader", function () {
...
+                    });
+                });
+            });
+            const maxIdx = assertionData["indexLOD"];
+            for (let i = 0; i <= maxIdx; i++) {
+                expect(assertionData[`setRequestHeaderCalls.${i}`]).toEqual(expectedSetRequestHeaderCalls.slice(0, 3 + i));
+            }
+            expect(assertionData["setRequestHeaderCalls2"]).toEqual(expectedSetRequestHeaderCalls);
+            // TODO - this fails! it has 1 more element than expected
+            // expect(assertionData["setRequestHeaderCalls3"]).toEqual(expectedSetRequestHeaderCalls.slice(0, 3));
```
Comment (RaananW) [inline packages/dev/loaders/test/integration/babylon.sceneLoader.test.ts:552]: @bghgary - for some reason the code above returns one more element in this array (should be 4 and not 3). I checked in logs as well, it is the correct behavior. Or at least the current behavior.

File: packages/dev/loaders/test/integration/babylon.sceneLoader.test.ts [hunk 2]
```diff
@@ -197,10 +197,960 @@ describe("Babylon Scene Loader", function () {
...
+                });
+            }, materialNames);
+
+            expect(Object.keys(assertionData)).toHaveLength(3);
+            materialNames.forEach((name) => {
+                expect(assertionData[name]).toEqual([name, name]);
+            });
+        });
+
+        // This test was deliberately left commented out as it is not working as expected.
```
Comment (RaananW) [inline packages/dev/loaders/test/integration/babylon.sceneLoader.test.ts:463]: @bghgary - I am not sure what was the purpose of this test, but the onReady function in the MSFT_lod extension generated promises that cannot be stopped, even if the loader is disposed (and null-ed...). So this test will continue failing until fixed. I left it commented out, but I would remove it if you say it is fine.

Comment (bghgary) [reply packages/dev/loaders/test/integration/babylon.sceneLoader.test.ts:463]: We need this test. It's one of the only tests covering the MSFT_lod extension.

**Review by bghgary — APPROVED** [review]
I think you can probably leave the two test that don't work commented out. File an issue for me to investigate and fix them.

File: packages/dev/loaders/test/integration/babylon.sceneLoader.test.ts (hunk 2 above)
Comment (RaananW) [reply packages/dev/loaders/test/integration/babylon.sceneLoader.test.ts:463]: The test right before it is also using MSFT_lod:

`Load LevelOfDetail with onMaterialLODsLoadedObservable`

Comment (RaananW) [reply packages/dev/loaders/test/integration/babylon.sceneLoader.test.ts:463]: The onReady function of the extension needs to be revised for this test to pass correctly

**Pull request merged**